core/src/tile/tileManager.cpp

```diff
@@ -14,16 +14,18 @@
 
 namespace Tangram {
 
-TileManager::TileManager(TileTaskQueue& _tileWorker) : m_workers(_tileWorker) {
+TileManager::TileManager(std::shared_ptr<TileTaskQueue> _tileWorker) : m_workers(_tileWorker) {
 
     m_tileCache = std::unique_ptr<TileCache>(new TileCache(DEFAULT_CACHE_SIZE));
 
     m_dataCallback = TileTaskCb{[this](std::shared_ptr<TileTask>&& task) {
+        if (m_workers) {
             if (task->hasData()) {
-                m_workers.enqueue(std::move(task));
+                m_workers->enqueue(std::move(task));
             } else {
                 task->cancel();
             }
```

Comment:
That's a good suggestion. Not to make the callback happen if the task was canceled at the first place. You might need a mutex to sync the task cancelation update though?